TLF-74: Creating a pipeline for LCS

DivineTaminangHO opened the pull request:
## What?

This is a pipeline for landlords checking service. It's purpose is for landlords to check all adult tenants for rights to rent in the UK.
It includes a .drone.yaml file for the the Drone pipeline, and Quay repository
 Trivy, Snyk, SonarQube and Cron Jobs for tearing down envs will be added to the source code
Trivy detects vulnerabilities of OS packages, Docker Images and application dependencies. Snyk doesn't scan Docker Images but with webUI we can easily find out the vulnerabilities. It also scans the Kubernetes manifest files. SonarQube is a Code Quality Assurance tool that collects and analyzes source code, and provides reports for the code quality of your project

##Why?

The aim of this is to enable landlords to check if their adult tenants have the right to rent in UK.
And this pipeline will ensure that the code is build and pushed to quay repo by automation

## How?

I have build a drone pipeline consisting of Github for version control, Drone CI for continuous integration and code build and the trivy, snyk and sonarQube scanners will scan the code.
## Testing?

The pipeline will be tested end-to-end when it's completed.
## Screenshots (optional)

## Screenshots (optional)
<img width="1440" alt="Screenshot 2024-04-15 at 22 38 41" src="https://github.com/UKHomeOffice/landlords-checking-service/assets/162344062/7a1e9434-0d35-44f8-8b00-22116e1aa00a">


<img width="1384" alt="image" src="https://github.com/UKHomeOffice/landlords-checking-service/assets/162344062/2b0fccd6-0480-4e17-b1c7-d7376e6344d4">


## Check list

- [ ] I have reviewed my own pull request
- [ ] I have written tests (if relevant)

DivineTaminangHO committed:
new branch created to test pipeline

DivineTaminangHO committed:
updated drone files to test pipeline

DivineTaminangHO committed:
updated drone files to test pipeline

DivineTaminangHO committed:
updated drone files to test pipeline

DivineTaminangHO committed:
updated drone files to test pipeline

DivineTaminangHO committed:
updated drone files to test pipeline

DivineTaminangHO committed:
changed some files from lmr to lcs

adityababumallisettiHO committed:
Update README.md

DivineTaminangHO committed:
added changes for test purpose

DivineTaminangHO committed:
Update README.md

DivineTaminangHO committed:
Update README.md

DivineTaminangHO committed:
Update .drone.yml

DivineTaminangHO committed:
Update .drone.yml

DivineTaminangHO committed:
updated readme.md

DivineTaminangHO committed:
updated .drone.yml

DivineTaminangHO committed:
updated yarn.lock files

DivineTaminangHO committed:
yarn.lock updated

DivineTaminangHO committed:
yarn.lock updated

DivineTaminangHO committed:
pipeline check

DivineTaminangHO committed:
pipeline check

DivineTaminangHO committed:
*LCS-74: DRONE PIPELINE LANDLORDS CHECKING SERVICES

*Pipeline created for lcs

*Creation of Github repo files, drone ci configurations, quay repo  and snyk scanners

*Kubernetes secrets for the uat and branch namespaces created

yarn.lock

sulthan-ahmed commented:
Hi @DivineTaminangHO nice work.  There are some standards we have agreed as a team when we do our PRs, can you have a look at our documentation in the team and update your PR in relation to it.  Check out @adityababumallisettiHO PRs as great examples of how to do this https://github.com/UKHomeOfficeForms/hof-services-config/pull/209

DivineTaminangHO committed:
*TLF-74: DRONE PIPELINE LANDLORDS CHECKING SERVICES
*Pipeline created for lcs
*Creation of Github repo files, drone ci configurations, quay repo  and snyk scanners
*Kubernetes secrets for the uat and branch namespaces created
*pipeline updated to add snyk to .drone.yml

DivineTaminangHO committed:
added snyk test to package.json

DivineTaminangHO committed:
added snyk test to package.json

DivineTaminangHO committed:
added snyk test to package.json

DivineTaminangHO committed:
yarn.lockupdated

DivineTaminangHO committed:
yarn.lockupdated

DivineTaminangHO committed:
yarn.lockupdated

DivineTaminangHO committed:
.snyk updated

sulthan-ahmed commented on .drone.yml:
```diff
@@ -0,0 +1,320 @@
+#         include:
+#           - master
+#           - feature/*
+#       event: pull_request
```
is there a reason this is commented out?

sulthan-ahmed commented on .drone.yml:
```diff
@@ -0,0 +1,320 @@
+      - npm run test:acceptance
+    when:
+      branch: master
+      event: push
```
is playwright being used or will be used

sulthan-ahmed commented:
This has got so much in this PR, it has actual code as well pipeline work

DivineTaminangHO committed:
Added CVE's to snyk to ignore common vulnerabilities

DivineTaminangHO committed:
Added CVE's to snyk to ignore common vulnerabilities

DivineTaminangHO committed:
Added CVE's to snyk to ignore common vulnerabilities

DivineTaminangHO committed:
Added CVE's to snyk to ignore common vulnerabilities

DivineTaminangHO committed:
Added CVE's to snyk to ignore common vulnerabilities

DivineTaminangHO commented:
Hi @sulthan-ahmed, I have removed the files which needs to be added by developers but I did no delete the files which are common for all services. Can you please review ?

DivineTaminangHO committed:
TLF-74: Removed irrelevant files

* Created a pipeline for LCS
* Removed all dev files
* This files will be added by developers

DivineTaminangHO committed:
TLF-74: Removed irrelevant files

* Created a pipeline for LCS
* Removed all dev files
* This files will be added by developers

sulthan-ahmed commented on README.md:
```diff
@@ -0,0 +1,29 @@
+- NPM (installed with Node.js) - Works with version 18+
+- [Html-pdf-converter](https://github.com/UKHomeOffice/html-pdf-converter) service running on a separate port to this service
+- [Redis server](http://redis.io/download) running on the default port
+
```
maybe delete all of the below in the ReadME as that is more for a dev to write about env variables and I'm not sure what is needed

sulthan-ahmed commented on config.js:
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
```
this file should be done by a developer

DivineTaminangHO committed:
TLF-74: Removed irrelevant files

* Created a pipeline for LCS
* Removed all dev files
* This files will be added by developers

gadorcruz commented on kube/certs/certificate-external.yml:
```diff
@@ -0,0 +1,14 @@
+metadata:
+  name: branch-tls-external
+  labels:
+    cert-manager.io/solver: route53
```
Have ACP allowed access through route53? Otherwise http01 is needed:
https://docs.acp.homeoffice.gov.uk/how-to/security/cert-manager/

gadorcruz commented on .drone.yml:
```diff
@@ -0,0 +1,296 @@
+  APP_NAME: lcs
+  PROD_ENV: sas-lcs-prod
+  BRANCH_ENV: sas-lcs-branch
+  STG_ENV: sas-lcs-stg
```
The staging environment for the project is not needed and might need to be removed from the files

gadorcruz commented:
The pipeline was created through another branch
